numpy/compat/py3k.py

```diff
@@ -53,6 +53,9 @@ def asstr(s):
 def isfileobj(f):
     return isinstance(f, (io.FileIO, io.BufferedReader, io.BufferedWriter))
```

Comment:
I wonder whether the better fix is to just change this line to:

```suggestion
    return isinstance(f, (io.RawIOBase, io.BufferedIOBase))
```

Reply:
CC: @rossbar 

If I change the implementation of `isfileobj` to allow BytesIO to pass through, i.e make it
`return hasattr(f, 'readinto')` or `return isinstance(f, (io.RawIOBase, io.BufferedIOBase))`, then its usage in 
`np.lib.format.write_array` and `np.lib.format.read_array` becomes buggy [here](https://github.com/numpy/numpy/blob/b9c6f3b28c83b3965f54bcc37331a94f04bfe6b4/numpy/lib/format.py#L686):
The following code then throws an error (which it shouldnt):

```python
import numpy as np
from io import BytesIO

bytes_array = BytesIO()
np.lib.format.write_array(bytes_array, x)

Traceback (most recent call last):                                                                                  
  File "<stdin>", line 1, in <module>                                                                               
  File ".../numpy/numpy/lib/format.py", line 677, in write_array                                  
    array.tofile(fp)                                                                                                
io.UnsupportedOperation: fileno
```

Reply:
Ah, I see. So if I understand correctly, there's no way to re-implement `isfileobj` to return True for BytesIO without breaking compatibility? Or is there just something strange in the `lib.format` functions?

Reply:
>there's no way to re-implement isfileobj to return True for BytesIO without breaking compatibility? 

Yes, that is indeed the case. If it returns True for BytesIO then we will break compatibility in `lib.format`.

Also `lib.format` is the only place now where `isfileobj` is used. So we could remove this method with its impl. directly at the caller. Is it something that we would want?